snuba/datasets/dataset.py

```diff
@@ -88,6 +91,23 @@ def get_abstract_columnset(self) -> ColumnSet:
         # TODO: Make this available to the dataset query processors.
         return self.__abstract_column_set
 
+    def get_column_resolver(self) -> ColumnResolver:
+        """
+        Returns the component that resolves the columns referenced in the query as strings in
+        columns present in the logical schema defined by get_abstract_columnset, thus identifying
+        the table and the structure of the columns.
+
+        The resolver is provided by the dataset and not by the Schema or by the ColumnSet because
+        they are still shared abstractions we use for both the logical schema and the physical schema
+        (which should change). A ColumnResolver is a concept related to the logical schema only,
+        so a storage schema should not provide it.
+
+        TODO: Entities are likely to change the structure of the logical schema, which is just
+        a ColumnSet now. Then we should be able to have only one implementation of ColumnResolver
+        that is built out of the logical schema.
+        """
+        return self.__column_resolver
```

Comment:
I think, when we introduce entities, we will necessarily have to redesign the data structure that represents the logical schema. Now we just use a ColumnSet, basically because there was no time to redesign it yet and none uses it.
With entities I would expect we branched off from ColumnSet (which should be used to define a clickhouse table schema instead) and have a structured logical data model that can represent either joined tables or single tables. 
If/when that happens the dataset. will not need anymore to return the resolver (which I don't like because it is coupled to the query language). Instead the build_query function, after parsing, will build a ColumnResolver instantiating it with the logical schema. There will be only one implementation of ColumnResolver around.

Something like
```
resolve = ColumnResolver(dataset.get_abstract_schema())
# now resolve the columns
```
At this point the dataset will not need to know what a ColumnResolver is.